Adds a new taar_amowhitelist command and ETL job to mozetl

crankycoder opened the pull request:
This adds a new taar_amowhitelist command to fetch the AMO database dump in taar_amodump, filter it down and upload a AMO addon whitelist.

Dexterp37 commented on mozetl/taar/taar_amowhitelist.py:
```diff
@@ -0,0 +1,86 @@
+logger = logging.getLogger('amo_whitelist')
+
+
+class AMOTransformer:
```
Can we please add a documentation for this? What is it supposed to do? For which purpose?

crankycoder replied:
Docs added

Dexterp37 commented on mozetl/taar/taar_amowhitelist.py:
```diff
@@ -0,0 +1,86 @@
+        self._min_age = min_age
+
+    def extract(self):
+        with NamedTemporaryFile() as dst_json_file:
```
The same can be achieved with something like `json.loads(s3.Object(BUCKET_NAME, S3_KEY).get()['Body'])` without resorting to a temporary file. Just FYI if you want to use that rather than this.

crankycoder replied:
Thanks!  I've put that code into the taar_utils.read_from_s3 function now.

Dexterp37 commented on mozetl/taar/taar_amowhitelist.py:
```diff
@@ -0,0 +1,86 @@
+            data = open(dst_json_file.name).read()
+            return json.loads(data)
+
+    def transform(self, json_data):
```
Let's document this: what are the criteria here? Why are we using them? Was this discussed somewhere that can be referenced from the code (e.g. link?)?

crankycoder replied:
Comments and links to the issue added now.

Dexterp37 commented on tests/test_taar_utils.py:
```diff
@@ -42,6 +42,31 @@
+
+        data = open(dst_json_file.name).read()
+        assert data == json.dumps(SAMPLE_DATA)
+
```
Do you want to test what happens if the URI is invalid/the file is not available as well?

crankycoder replied:
I don't think that's necessary.  These are going to be ETL jobs and airflow will just flag us for errors.  If an upstream datasource is broken, I think it's the responsibility of that upstream job to correct the issue.

crankycoder commented:
@Dexterp37 i think i've addressed all issues in the review except maybe https://github.com/mozilla/python_mozetl/pull/216#discussion_r179082407 , but I think the scope of responsibility for failed upstream files should lie with upstream data providers.

mlopatka commented:
@Dexterp37 can we merge this in and switch over to the new white-list for all TAAR applications? 
I want to make sure this is being run via airflow before relying on it for downstream dependencies.
